### PR TITLE
T35078: Get all child nodes from parent

### PR DESCRIPTION
**Pull request by JenySadadia**
Send GET request to /trigger_completed_event endpoint of API.

### CODE REVIEW

File: kernelci/db/kernelci_api.py [hunk 1]
```diff
@@ -103,6 +103,16 @@ def get_root_node(self, node_id):
         resp = self._get('/'.join(['get_root_node', node_id]))
         return json.loads(resp.text)
 
+    def get_completed_event(self, node_id, hours=0, minutes=0, seconds=0):
+        """Get an event when all child nodes are complete of a given node"""
+        resp = self._get(f'trigger_completed_event/{node_id}?wait_time_hours=\
+{hours}&wait_time_minutes={minutes}&wait_time_seconds={seconds}')
```
Comment (mgalka) [inline kernelci/db/kernelci_api.py:109]: When using query parameters we should consider using params kyeword argument for requests.get(), which is used by _get method.
https://requests.readthedocs.io/en/latest/api/#requests.get

File: kernelci/db/kernelci_api.py [hunk 2]
```diff
@@ -108,6 +108,20 @@ def get_regressions(self, regression_name):
         resp = self._get('?'.join(['regressions', 'name=' + regression_name]))
         return json.loads(resp.text)
 
+    def get_completed_event(self, node_id, hours=0, minutes=0, seconds=0):
+        """Get an event when all child nodes are complete of a given node"""
+        params = {
+            'wait_time_hours': hours,
+            'wait_time_minutes': minutes,
+            'wait_time_seconds': seconds
+        }
```
Comment (gctucker) [inline kernelci/db/kernelci_api.py:117]: Sorry I didn't notice this before, and I see the API parts have been merged already.  I think we should remove this completely, we can't have a query that waits on the API side when we should just have a timer on the client side.

**Review by gctucker — CHANGES_REQUESTED** [review]

File: kernelci/db/kernelci_api.py (hunk 2 above)
Comment (JenySadadia) [reply kernelci/db/kernelci_api.py:117]: So what I need here is to move the logic of `Node.wait_for_node` in the pipeline.
The endpoint will just check whether the node is completed or not and trigger event.

Comment (JenySadadia) [reply kernelci/db/kernelci_api.py:117]: Setting node status to `timeout` will be handled by `set_timeout` script.
I will update the `trigger_completed` pipeline script to wait for the child nodes to get completed.
When all child nodes are completed, the root status will be set to completed and the API endpoint will be called only to trigger `completed` event at the end.

**Comment by JenySadadia** [comment]
Removed methods related to `completed` event as we have discarded it from the API.

**Comment by JenySadadia** [comment]
Function from https://github.com/kernelci/kernelci-core/pull/1265 can be used to get child nodes from parent node id. Now we don't need any special method for this. Hence, closing this PR.

**Pull request closed without merging**